contracts/contracts/strategies/uniswap/UniswapV3Strategy.sol

```diff
@@ -280,27 +283,10 @@ contract UniswapV3Strategy is UniswapV3StrategyStorage {
     ) external override onlyVault nonReentrant {
         onlyPoolTokens(_asset);
 
-        IERC20 asset = IERC20(_asset);
-        uint256 selfBalance = asset.balanceOf(address(this));
-
-        if (selfBalance < amount) {
-            require(activeTokenId > 0, "Liquidity error");
-
-            // Delegatecall to `UniswapV3LiquidityManager` to remove
-            // liquidity from active LP position
-            // solhint-disable-next-line no-unused-vars
-            (bool success, bytes memory data) = address(_self).delegatecall(
-                abi.encodeWithSignature(
-                    "withdrawAssetFromActivePositionOnlyVault(address,uint256)",
-                    _asset,
-                    amount - selfBalance
-                )
-            );
-            require(success, "DelegateCall to close position failed");
-        }
+        require(activeTokenId == 0, "Close active position");
```

Comment:
curious what is the reason we want to enforce for the active position to be closed?

Reply:
Since Uniswap V3 will not be allowed to be set as default strategy, the only call to `.withdraw()` will be when the strategist is manually reallocating the funds. Closing the position (using the `closeActivePosition` method) gives more control over slippage. If we were to close the position here, we would have to either set `0` as slippage or complicate the code on Vault or the contract itself. It sounds like a good tradeoff to expect Strategist to close position before moving any funds